raven/src/main/java/org/opencadc/raven/ProtocolsGenerator.java

```diff
@@ -244,14 +250,28 @@ static void prioritizePullFromSites(List<StorageSite> storageSites) {
         return protos;
     }
 
+    static SortedSet<StorageSite> prioritizePushToSites(Set<StorageSite> storageSites, URI artifactURI,
+                                                        Map<URI, StorageSiteRule> siteRules) {
+        PrioritizingStorageSiteComparator comparator = new PrioritizingStorageSiteComparator(siteRules, artifactURI, null);
+        TreeSet<StorageSite> orderedSet = new TreeSet<>(comparator);
+        for (StorageSite storageSite : storageSites) {
+            if (storageSite.getAllowWrite()) {
+                orderedSet.add(storageSite);
+            }
+        }
+        // return the set in descending order, higher priority sites first.
+        return orderedSet.descendingSet();
```

Comment:
Why do you have to do this? Does the comparator order them backwards? Ugh.

Reply:
The comparator usually sorts ascending, higher ordered last in the set. The StorageSite ordering is also used, which would be ascending, so thought best to be consistent. But if the StorageSite ordering is being used the order isn't important. Changed the comparator to order descending for all but the default StorageSite ordering.